File: dbfit-java/core/src/main/java/dbfit/fixture/QueryStats.java
package dbfit.fixture;

import dbfit.api.DBEnvironment;
import dbfit.api.DbEnvironmentFactory;
import dbfit.api.DbEnvironmentFacade;
import dbfit.util.FitNesseTestHost;
import dbfit.util.DataTable;

import java.sql.SQLException;

public class QueryStats extends fit.ColumnFixture {
    private DbEnvironmentFacade environmentFacade;
    private boolean hasExecuted = false;
    private int _rows;

    public String tableName;
    public String where;
    public String query;

    public QueryStats() {
        this(DbEnvironmentFactory.getDefaultEnvironment());
    }

    public QueryStats(DBEnvironment environment) {
        this.environmentFacade = new DbEnvironmentFacade(environment,
                FitNesseTestHost.getInstance());
    }

    public void setViewName(String value) {
        tableName = value;
    }

    public void reset() {
        hasExecuted = false;
        where = null;
        query = null;
        _rows = 0;
        tableName = null;
    }

    private void execQuery() throws SQLException {
        if (hasExecuted) {
            return;
        }

        if (query == null) {
            query = "select * from " + tableName + (where != null ? " where " + where : "");
        }

        String countQuery = "select count(*) from (" + query + ") temp";
        DataTable dt = environmentFacade.getQueryTable(countQuery);
        String columnName = dt.getColumns().get(0).getName();
        _rows = ((Number) dt.getRows().get(0).get(columnName)).intValue();

        hasExecuted = true;
    }

    public int rowCount() throws SQLException {
        execQuery();
        return _rows;
    }

    public boolean isEmpty() throws SQLException {
        return rowCount() == 0;
    }
}

File: dbfit-dotnet/core/src/fixture/QueryStats.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Data;
using System.Data.Common;
namespace dbfit.fixture
{
    public class QueryStats:fit.ColumnFixture
    {
        private IDbEnvironment environment;
        public QueryStats()
        {
            environment = DbEnvironmentFactory.DefaultEnvironment;
        }
        public QueryStats(IDbEnvironment environment)
        {
            this.environment = environment;
        }
        public string TableName;
        public string ViewName { set { TableName=value;}}
        public string Where;
        public string Query;
        private bool hasExecuted=false;
        public override void Reset()
        {
            hasExecuted = false;
            Where = null;
            Query = null;
            _rows = 0;
            TableName = null;
        }
        private int _rows;
        private void execQuery()
        {
            if (hasExecuted) return;
            if (Query == null)
            {
                Query = "select count(*) from " + TableName + (Where != null ? " where " + Where : "");
                DbCommand dc = environment.CreateCommand(Query, CommandType.Text);
                object o=dc.ExecuteScalar();
                dc.Dispose();
                if (o != null) _rows = Convert.ToInt32(o);
            }
            else
            {
                DbCommand dc = environment.CreateCommand(Query, CommandType.Text);
                environment.BindFixtureSymbols(dc);

                DbDataAdapter oap = environment.DbProviderFactory.CreateDataAdapter();
                oap.SelectCommand = dc;
                DataSet ds = new DataSet();
                oap.Fill(ds);
                dc.Dispose();
                _rows=ds.Tables[0].Rows.Count;
            }
            hasExecuted = true;
        }
        public int RowCount()
        {
            execQuery();
            return _rows; 
        }
        public bool IsEmpty()
        {
            return RowCount()==0; 
        }
    }
}
